src/Spectre.libStatistics/Math.h

```diff
@@ -380,4 +380,30 @@ std::vector<DataType> build(size_t size, Generator generator)
     return result;
 }
 
+/// <summary>
+/// Calculates differences between adjacent elements of given container.
+/// </summary>
+/// <param name="data">The container.</param>
+/// <param name="order">Order of difference operation (how many times the difference is executed).</param>
+/// <returns>Vector of calculated differences.</returns>
+ template <class DataType>
+ std::vector<DataType> diff(gsl::span<const DataType> data, uint16_t order = 1)
+ {
+    size_t length = data.length();
+    std::vector<DataType> result = std::vector<std::remove_const<DataType>::type>(data.begin(), data.end());
+
+    if (order == 0 || length < 2 || order >= length)
+        return result;
+
+    for(auto i = 0u; i < order; ++i)
+    {
+        gsl::span<const DataType> lhs = gsl::as_span(result.data() + 1, result.size() - 1);
+        gsl::span<const DataType> rhs = gsl::as_span(result.data(), result.size() - 1);
+        memcpy(result.data(), std::vector<DataType>(minus(lhs, rhs)).data(), (result.size() - 1) * sizeof(DataType));
```

Comment:
We're allocating the memory either way. Why don't we simply do `result = minus(lhs, rhs)`? It would exercise `move` operator, without copying and resizes.

Reply:
Nuhuh, such assignment changes the position in memory of `result`. This approach preserves it.

Reply:
But we're allocating the memory in `minus` either way, while constant position of `result` in memory gives us no gain. Am I wrong?

Reply:
That is why I considered doing it in some other way than using `minus`. It is here mostly just for the same of using already created functionality. I will tinker with `transform`.